### PR TITLE
feat(data): add Nemotron Image v3 Energon tutorial

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do?

Adds a pinned, small-subset preparation and training tutorial for
`nvidia/Nemotron-Image-Training-v3` through the Energon path introduced by
#5285.

This is intentionally stacked on #5285. It does not download or materialize
the complete 6.9M-sample release.

## Contents

- Add `prepare_nemotron_image_v3.py` to join selected conversation JSONL files
  with separately stored media and write Bridge `ChatMLWebdataset` shards.
- Pin the dataset revision and verify the self-contained `turing` JSONL and
  media tar by size and SHA-256 before conversion.
- Restrict verified conversion to the pinned archive allowlist, reject
  unpinned subsets and loose/unverified media shadows, and provide an explicit
  `--skip-source-integrity-check` escape hatch.
- Normalize the mixed source message representation to typed ChatML, create a
  deterministic train/validation split, run Energon indexing, and persist a
  preparation manifest.
- Fail closed on invalid/remote/ambiguous media, unsupported video/audio,
  archive path escapes, stale output directories, and constructor-time archive
  failures without leaking open handles.
- Document a complete small-subset flow: pinned Hugging Face download,
  conversion/indexing, Qwen3-VL checkpoint import, and a public
  `scripts/training/train.sh` LoRA launch using `--dataset energon`, `vlm_step`,
  16K sequence length, and native Energon packing.
- Link the tutorial from the data, Energon, training, and Qwen3-VL entry points.
- Add focused happy-path and failure-path unit coverage, and restore collection
  of the existing `tests/unit_tests/tutorials/` suite.

## Validation

- Exact-head Core Unit job: `9044 passed, 14 skipped, 2 deselected`; both
  tutorial modules were collected, including all 13 Nemotron Image v3 cases.
- Real pinned `turing` conversion: 193 source conversations converted into 181
  train and 12 validation samples; source sizes and SHA-256 values matched,
  image bytes decoded, and Energon metadata/indexes were created.
- Real Qwen3-VL Energon loader: produced a finite multimodal batch with labels,
  loss mask, visual inputs, THD metadata, and a positive token count.
- Exact `train.sh` smoke: one 16K Qwen3-VL LoRA optimizer step plus validation
  completed with finite losses, zero skipped/NaN iterations, a persisted
  resolved config, and model plus Energon dataloader checkpoints.
- `uv run pre-commit run --all-files` passed in Linux.
- `ruff check`, `ruff format --check`, strict mypy, Python compilation, and
  `git diff --check` passed.

The child PR's exact-head Core Unit job now collects and passes the tutorial
suite. The remaining full-suite Ministral3 packing failure is inherited from
stacked parent #5285 and remains a parent integration gate.

### CODE REVIEW

**Comment by yaoyu-33** [comment]
Addressed the review gap in 5ae6f680: added direct tests for remote and ambiguous media references plus video/audio rejection. The follow-up also covers pinned-source missing/size/hash failures, unpinned and shadow media, invalid image references, archive escape/cleanup, stale output, and Energon indexing arguments. Commit 40dd4f93 restores collection of `tests/unit_tests/tutorials/`; the exact-head [Core Unit job](https://github.com/NVIDIA-NeMo/Megatron-Bridge/actions/runs/31362668506/job/93378970511) now runs both tutorial modules and passes with 9044 passed / 14 skipped / 2 deselected, including all 13 Nemotron-specific cases.

**Comment by yaoyu-33** [comment]
/ok to test https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/5455/commits/75f1ed5356acbe2e8a0d58747ea6b1b6e425947b

File: tutorials/data/energon/nemotron-image-v3.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  dataset.packing_buffer_size=16 \
+  dataset.task_encoder.hf_processor_revision="$MODEL_REVISION" \
+  +tokenizer.hf_tokenizer_kwargs.revision="$MODEL_REVISION" \
+  model.recompute_granularity=full \
+  model.recompute_method=uniform \
+  model.recompute_num_layers=1 \
+  model.calculate_per_token_loss=True \
+  ddp.average_in_collective=False \
+  logger.log_interval=1 \
+  logger.save_config_filepath="$OUTPUT_DIR/resolved-config.yaml"
```
Comment (cuichenx) [inline tutorials/data/energon/nemotron-image-v3.md:140]: is it possible to not have so many overrides? only include the necessary ones for this example

File: tutorials/data/energon/prepare_nemotron_image_v3.py
Comment (cuichenx) [inline tutorials/data/energon/prepare_nemotron_image_v3.py:1]: nit, there's some duplication with other energon preparation scripts, possible to consolidate?

**Review by cuichenx — DISMISSED** [review]
two minor comments

File: tutorials/data/energon/nemotron-image-v3.md (hunk 1 above)
Comment (yaoyu-33) [reply tutorials/data/energon/nemotron-image-v3.md:140]: Addressed in a8fc234d40a1c2fdced79e549dad50d4badbc75f. The command now relies on the exact recipe for LoRA, Energon, and vlm_step, and on the launcher default for one node. I also removed overrides whose values are derived/defaulted (validation micro-batch size, validation worker count, and the separate tokenizer revision). The remaining packed-run requirements are documented separately from the one-step smoke/debug settings. Focused recipe/launcher tests and all-files pre-commit pass.

File: tutorials/data/energon/prepare_nemotron_image_v3.py
Comment (yaoyu-33) [reply tutorials/data/energon/prepare_nemotron_image_v3.py:1]: Addressed in a8fc234d40a1c2fdced79e549dad50d4badbc75f. I removed the local SplitShardWriter and now use webdataset.ShardWriter, which is already used by an existing Energon preparation tutorial. The small wrapper only fixes deterministic tar metadata; writers are created lazily so a split with zero samples does not produce an empty shard. A focused synthetic conversion smoke verified shard rollover, member names/payloads, deterministic metadata, and the no-empty-validation-shard behavior; all-files pre-commit also passes.
